.github/workflows/build_rocks.yaml

```diff
@@ -0,0 +1,107 @@
+name: Build images
+
+on:
+  workflow_call:
+    inputs:
+      owner:
+        type: string
+        description: Registry owner to push the built images
+        default: ""
+      registry:
+        type: string
+        description: Registry to push the built images
+        default: ""
+      runs-on:
+        type: string
+        description: Image runner for building the images
+        default: ubuntu-22.04
+      trivy-image-config:
+        type: string
+        description: Trivy YAML configuration for image testing that is checked in as part of the repo
+      working-directory:
+        type: string
+        description: The working directory for jobs
+        default: "./"
+    outputs:
+      images:
+        description: List of images built
+        value: ${{ jobs.get-rocks.outputs.rocks }}
+
+jobs:
+  get-rocks:
+    name: Get rocks
+    runs-on: ubuntu-22.04
+    outputs:
+      rock-paths: ${{ env.ROCK_PATHS }}
+      rocks: "${{ env.ROCKS }}"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get rock paths
+        working-directory: ${{ inputs.working-directory }}
+        run: |
+          lines=$(find . -type f -name rockcraft.yaml | wc -l)
+          if [ $lines -ne 0 ]; then
+            echo "ROCK_PATHS=$(find . -type f -name rockcraft.yaml | xargs realpath | xargs dirname | jq -Rsc '. / "\n" - [""]')" >> $GITHUB_ENV
+            echo "ROCKS=$(find . -type f -name rockcraft.yaml | xargs -l yq '.name' | jq -Rsc '. / "\n" - [""]')" >> $GITHUB_ENV
+          else
+            echo "ROCK_PATHS=$(echo "" | jq -Rsc '. / "\n" - [""]')" >> $GITHUB_ENV
+            echo "ROCKS=$(echo "" | jq -Rsc '. / "\n" - [""]')" >> $GITHUB_ENV
+          fi
+  build-rocks:
+    name: Build rock
+    runs-on: ${{ inputs.runs-on }}
+    needs: [get-rocks]
+    if: ${{ needs.get-rocks.outputs.rock-paths != '[]' }}
+    strategy:
+      matrix:
+        path: ${{ fromJSON(needs.get-rocks.outputs.rock-paths) }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build rock
+        uses: canonical/craft-actions/rockcraft-pack@main
+        with:
+          path: ${{ matrix.path }}
+      - name: Upload ROCK to ghcr.io
+        if: ${{ !github.event.pull_request.head.repo.fork }}
+        run: |
+          IMAGE_NAME=$(yq '.name' ${{ matrix.path }}/rockcraft.yaml)
+          IMAGE_REF=${{ inputs.registry }}/${{ inputs.owner }}/$IMAGE_NAME:${{ github.run_id }}
+          skopeo --insecure-policy copy oci-archive:$(ls ${{ matrix.path }}/${{ matrix.image }}*.rock) docker://$IMAGE_REF --dest-creds "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}"
```

Comment:
No risks of catching multiple files with `${{ matrix.path }}/${{ matrix.image }}*.rock` ?

Reply:
No, at the moment only one descriptor named rockcraft.yaml is allowed so we'll only be getting one *.rock per directory https://github.com/canonical/rockcraft/issues/185

Reply:
I'm wondering if you won't have multiple .rock file if there are multiple platforms supported in the rockcraft.yaml

Reply:
Not for out use cases. The same would be applicable to the file resulting from packing the charm

Reply:
Yes I agree, I don't want to address this case, but if it happens this workflow will fail because the $(ls...) will return multiple files
So I would at list restrict to the first result of the `ls` to make sure the workflow doesn't just fail

Reply:
It would be easier to catch an issue if it fails when compared to using an image not expected by the user, IMHO

Reply:
Ok we'll see how to manage that if someday it breaks due to multiple platform